File: MeusLivros/app/src/main/java/br/com/senai/meuslivros/MainActivity.java
package br.com.senai.meuslivros;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.SearchView;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;

import java.util.List;

import br.com.senai.meuslivros.adapter.LivroAdapter;
import br.com.senai.meuslivros.dao.LivroDAO;
import br.com.senai.meuslivros.model.Livro;

public class MainActivity extends AppCompatActivity {

    private SearchView searchView;
    private RecyclerView listaLivros;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        searchView = findViewById(R.id.campoBusca);
        listaLivros = findViewById(R.id.listaDeLivros);
        LivroDAO dao = new LivroDAO(this);
        List<Livro> livros = dao.buscaLivros();



        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, FormularioActivity.class);
                startActivity(intent);
            }
        });
    }

    private void carregarLista(List<Livro> livros) {
        LivroAdapter adapter = new LivroAdapter(this, livros);
        listaLivros.setAdapter(adapter);
        RecyclerView.LayoutManager layoutManager = new GridLayoutManager(this, 2);
        listaLivros.setLayoutManager(layoutManager);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }
        if (id == R.id.action_exit){
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onResume() {
        super.onResume();
        LivroDAO dao = new LivroDAO(this);
        carregarLista(dao.buscaLivros());
    }
}

File: UsandoFragments/app/src/main/java/informatica/sp/senai/br/usandofragments/MainActivity.kt
package informatica.sp.senai.br.usandofragments

import android.os.Bundle
import android.support.v4.app.FragmentTransaction
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuInflater
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val botaoTrocar = findViewById<Button>(R.id.btnTrocarFragments)
        var status = true
        botaoTrocar.setOnClickListener({
            var fragmentTransaction: FragmentTransaction = supportFragmentManager.beginTransaction()
            if (status) {
                var loginFragment: LoginFragment = LoginFragment()
                fragmentTransaction.replace(R.id.container_fragments, loginFragment)
                fragmentTransaction.commit()
                botaoTrocar.text = "Cadastro"
                status = false
            } else {
                var cadastroFragment: CadastroFragment = CadastroFragment()
                fragmentTransaction.replace(R.id.container_fragments, cadastroFragment)
                fragmentTransaction.commit()
                botaoTrocar.text = "Login"
                status = true
            }
        })

    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        var menuInflater: MenuInflater = menuInflater
        menuInflater.inflate(R.menu.menu_principal, menu)
        return true
    }
}
